module-rt/tests/multiple.rs

```diff
@@ -0,0 +1,213 @@
+use remote_trait_object::{service, Config as RtoConfig, Context as RtoContext, Service, ServiceRef, ServiceToImport};
+use std::sync::Arc;
+
+#[service]
+trait Pizza: Service {}
+
+#[service]
+trait PizzaBox: Service {}
+
+#[service]
+trait PizzaStore: Service {
+    fn create_pizza(&self) -> ServiceRef<dyn Pizza>;
+    fn wrap_pizza_box(&self) -> ServiceRef<dyn PizzaBox>;
+}
+
+struct SimplePizza;
+impl Service for SimplePizza {}
+impl Pizza for SimplePizza {}
+
+struct SimplePizzaBox {
+    _pizzas: Vec<Box<dyn Pizza>>,
+}
+impl Service for SimplePizzaBox {}
+impl PizzaBox for SimplePizzaBox {}
+
+struct SimplePizzaStore {
+    pizza_pool: Arc<RwLock<Vec<Box<dyn Pizza>>>>,
+}
+impl Service for SimplePizzaStore {}
+impl PizzaStore for SimplePizzaStore {
+    fn create_pizza(&self) -> ServiceRef<dyn Pizza> {
+        ServiceRef::create_export(Box::new(SimplePizza) as Box<dyn Pizza>)
+    }
+
+    fn wrap_pizza_box(&self) -> ServiceRef<dyn PizzaBox> {
+        let mut rng = rand::thread_rng();
+        let mut pool = self.pizza_pool.write();
+        let n = pool.len() - rng.gen_range(0, std::cmp::min(5, pool.len()));
+        let pizzas = pool.split_off(n);
+        ServiceRef::create_export(Box::new(SimplePizzaBox {
+            _pizzas: pizzas,
+        }) as Box<dyn PizzaBox>)
+    }
+}
+struct ModuleA {
+    pizza_stores: Vec<Box<dyn PizzaStore>>,
+    pizza_pool: Arc<RwLock<Vec<Box<dyn Pizza>>>>,
+    pizza_boxes: Vec<Box<dyn PizzaBox>>,
+}
+
+impl UserModule for ModuleA {
+    fn new(_arg: &[u8]) -> Self {
+        Self {
+            pizza_stores: Default::default(),
+            pizza_pool: Default::default(),
+            pizza_boxes: Default::default(),
+        }
+    }
+
+    fn prepare_service_to_export(&mut self, _ctor_name: &str, _ctor_arg: &[u8]) -> Skeleton {
+        Skeleton::new(Box::new(SimplePizzaStore {
+            pizza_pool: Arc::clone(&self.pizza_pool),
+        }) as Box<dyn PizzaStore>)
+    }
+
+    fn import_service(
+        &mut self,
+        rto_context: &RtoContext,
+        _exporter_module: &str,
+        _name: &str,
+        handle: HandleToExchange,
+    ) {
+        self.pizza_stores.push(import_service_from_handle(rto_context, handle));
+    }
+
+    fn debug(&mut self, _arg: &[u8]) -> Vec<u8> {
+        let mut rng = rand::thread_rng();
+        let pizza_n = rng.gen_range(100, 1000);
+        for _ in 0..pizza_n {
+            self.pizza_pool
+                .write()
+                .push(self.pizza_stores.choose(&mut rng).unwrap().create_pizza().unwrap_import().into_proxy());
+        }
+
+        let pizza_box_n = rng.gen_range(10, 100);
+        let mut pizza_boxes: Vec<Box<dyn PizzaBox>> = Vec::new();
+        for _ in 0..pizza_box_n {
+            pizza_boxes.push(self.pizza_stores.choose(&mut rng).unwrap().wrap_pizza_box().unwrap_import().into_proxy())
+        }
+        self.pizza_boxes = pizza_boxes;
+
+        Vec::new()
+    }
```

Comment:
It's hard to understand what is happening in this test.
Please write comments that explain the scenario.

Reply:
updated